cmake/modules/AddSwift.cmake

```diff
@@ -371,6 +371,10 @@ function(_add_host_variant_link_flags target)
       ${SWIFT_HOST_VARIANT_ARCH}_LIB)
     target_link_directories(${target} PRIVATE
       ${${SWIFT_HOST_VARIANT_ARCH}_LIB})
+  elseif(SWIFT_HOST_VARIANT_SDK STREQUAL WASI)
+    # WASI lacks a true mmap; enabling minimal mmap emulation
+    # see https://github.com/WebAssembly/wasi-libc/blob/86550c37ab14962233f73413ea0e0c0e57d056d7/libc-top-half/musl/include/sys/mman.h#L3
+    list(APPEND result "-Wl,wasi-emulated-mman")
```

Comment:
I don't think it makes sense to build the toolchain for WASI.  Why does the toolchain need to be built for WASI?

Reply:
It makes a huge amount of sense to build the toolchain for WASI (not that anyone has done so today, AFAIK).

For one thing, it would allow you to run the compiler in your browser. Very cool, and a great learning tool. Another use-case would be to bootstrap building the compiler as projects like Swift-driver make progress and more Swift is used in the compiler itself.